spyder/plugins/variableexplorer/widgets/main_widget.py

```diff
@@ -406,6 +450,20 @@ def switch_widget(self, nsb, old_nsb):
     # ---- Public API
     # ------------------------------------------------------------------------
 
+    def change_filter_state(self, value):
+        """Handle the change of the filter state."""
+        self.filter_on = not self.filter_on
+        self.filter_button.setChecked(self.filter_on)
+        self.filter_button.setToolTip(_("Filter variables"))
+        self.filter_variables()
+
+    def filter_variables(self):
```

Comment:
Let's rename this method to `_enable_filter_actions` and move it to the Private API section.